src/lib.rs

```diff
@@ -1413,7 +1443,7 @@ extern "C" fn fn_verify_init(
     let mech = res_or_ret!(token.get_mechanisms().get(data.mechanism));
     if mech.info().flags & CKF_VERIFY == CKF_VERIFY {
         let operation = res_or_ret!(mech.verify_new(data, &obj));
-        session.set_operation(Operation::Verify(operation));
+        session.set_operation(Operation::Verify(operation), obj.always_auth());
```

Comment:
Same here. The `CKA_ALWAYS_AUTHENTICATE` is defined only for private key objects:

https://docs.oasis-open.org/pkcs11/pkcs11-base/v3.0/os/pkcs11-base-v3.0-os.html#_Toc29976581

And this operation is done on the public key object so this should be
```suggestion
        session.set_operation(Operation::Verify(operation), false);
```